### PR TITLE
[doc] Add LMI Text Embedding Inference user guide

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: serving/docs/lmi/user_guides/embedding-user-guide.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+The inference process involves:
+
+1. **Loading a Model**: Loading a model from local directory, S3 or from huggingface repository,.
+2. **Tokenization**: Breaking down the input text into tokens that the model can understand.
+3. **Embeddings**: Passing the tokens through the model to produce embeddings. Embedding is a multi-dimension vector that could be used for RAG or general embedding search.
+
+LMI supports Text Embedding Inference with the following engines:
+
+- OnnxRuntime
+- Candle
```
Comment (frankfliu) [inline serving/docs/lmi/user_guides/embedding-user-guide.md:14]: Our engine name is "Rust"

File: serving/docs/lmi/user_guides/embedding-user-guide.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+- TensorRT
+- Python
+
+Currently, the OnnxRuntime engine provides the best performance for text embedding in LMI. 
+
+## Supported Model Architectures
+
+The following text models are supported:
+
+- Bert (`BAAI/bge-base-en-v1.5`, `BAAI/bge-small-en-v1.5`, `BAAI/bge-large-en-v1.5`, etc.)
```
Comment (frankfliu) [inline serving/docs/lmi/user_guides/embedding-user-guide.md:24]: Why only BERT?, I think this is just restriction for Rust engine.

File: serving/docs/lmi/user_guides/embedding-user-guide.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+Other embedding models are also supported, but may not be as performant as the model architectures listed above.
+
+## Quick Start Configurations
+
+You can leverage LMI Text Embedding inference using the following starter configurations:
+
+### serving.properties
```
Comment (frankfliu) [inline serving/docs/lmi/user_guides/embedding-user-guide.md:32]: We should not promote this usage, we should guide user to:
1. Use djl:// model if the model already exist
2. use HF_MODEL_ID to convert the model at runtime
3. Manually import the model into DJL model format with our djl_convert/djl_import tool
4. and finally fully customize with `serving.properties`

File: serving/docs/lmi/user_guides/embedding-user-guide.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+```
+OPTION_ENGINE=OnnxRuntime
+HF_MODEL_ID=<your model id>
+OPTION_BATCH_SIZE=32
+```
+
+You can follow [this example](../deployment_guide/deploying-your-endpoint.md#configuration---environment-variables) to deploy a model with environment variable configuration on SageMaker.
+
+## Example
```
Comment (frankfliu) [inline serving/docs/lmi/user_guides/embedding-user-guide.md:53]:
```suggestion
## Deploy model to SageMaker
```

File: serving/docs/lmi/user_guides/embedding-user-guide.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+**HF_MODEL_TRUST_REMOTE_CODE**
+
+If the model artifacts contain custom modeling code, you should set this to true after validating the custom code is not malicious.
+If you are using a HuggingFace Hub model id, you should also specify `HF_REVISION` to ensure you are using artifacts and code that you have validated.
+
+This is an optional config, and defaults to `False`.
+
+**OPTION_ENGINE**
+
+This option represents the Engine to use, values include OnnxRuntime, TensorRT, Candle, etc.
```
Comment (frankfliu) [inline serving/docs/lmi/user_guides/embedding-user-guide.md:134]:
```suggestion
This option represents the Engine to use, values include `OnnxRuntime`, `PyTorch`, `Rust`, etc.
```

File: serving/docs/lmi/user_guides/embedding-user-guide.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+**OPTION_BATCH_SIZE**
+
+This option represents the dynamic batch size.
+
+This is an optional config, and defaults to `1`.
```
Comment (frankfliu) [inline serving/docs/lmi/user_guides/embedding-user-guide.md:140]: It worth the mention `SERVING_MIN_WORKERS` and `SERVING_MAX_WORKERS` for GPU case. And encourage user to only set them on GPU

File: serving/docs/lmi/user_guides/embedding-user-guide.md (hunk 1 above)
Comment (xyang16) [reply serving/docs/lmi/user_guides/embedding-user-guide.md]: Changed

File: serving/docs/lmi/user_guides/embedding-user-guide.md (hunk 3 above)
Comment (xyang16) [reply serving/docs/lmi/user_guides/embedding-user-guide.md:32]: If the user give HF_MODEL_ID, then the code will automatically convert for them.

File: serving/docs/lmi/user_guides/embedding-user-guide.md (hunk 2 above)
Comment (xyang16) [reply serving/docs/lmi/user_guides/embedding-user-guide.md:24]: Changed.

File: serving/docs/lmi/user_guides/embedding-user-guide.md (hunk 5 above)
Comment (xyang16) [reply serving/docs/lmi/user_guides/embedding-user-guide.md]: Done.

File: serving/docs/lmi/user_guides/embedding-user-guide.md (hunk 6 above)
Comment (xyang16) [reply serving/docs/lmi/user_guides/embedding-user-guide.md:140]: Changed.

File: serving/docs/lmi/user_guides/embedding-user-guide.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+    sagemaker_session=session,
+    serializer=serializers.JSONSerializer(),
+    deserializer=deserializers.JSONDeserializer(),
+)
+
+# Make an inference request against the endpoint
+predictor.predict(
+    {"inputs": "What is Deep Learning?"}
+)
+```
```
Comment (lanking520) [inline serving/docs/lmi/user_guides/embedding-user-guide.md:126]: could we also have a API schema doc defined here? Need to tell customer what is the expected output format

**Review by frankfliu — APPROVED** [review]

File: serving/docs/lmi/user_guides/embedding-user-guide.md (hunk 7 above)
Comment (xyang16) [reply serving/docs/lmi/user_guides/embedding-user-guide.md:119]: Done

File: serving/docs/lmi/user_guides/embedding-user-guide.md [hunk 8]
```diff
@@ -0,0 +1,981 @@
...
+  0.7707857,
+  0.33682466,
+  0.6252336,
+  0.9911188,
+  0.18310043,
+  -0.5297898,
+  1.1664978,
+  0.79496926,
+  0.28931668,
+  1.2245488,
```
Comment (lanking520) [inline serving/docs/lmi/user_guides/embedding-user-guide.md:839]: this is too long, could we trim into ... in the middle? And also show the expected shape

**Pull request merged**